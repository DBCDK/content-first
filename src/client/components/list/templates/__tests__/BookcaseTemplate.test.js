import React from 'react';
import BookcaseTemplate from '../BookcaseTemplate.component';
import renderer from 'react-test-renderer';

jest.mock('../SimpleList.component', () => 'SimpleList');
jest.mock('../../../bookcase/BookcaseItem.component', () => 'BookcaseItem');

const createTestElement = id => {
  return {
    book: {
      pid: 'pid' + id,
      title: 'some title' + id,
      creator: 'some creator' + id
    },
    links: {
      cover: '/cover' + id
    },
    position: {x: 0, y: 0},
    description: 'some description' + id
  };
};

const profile = {
  name: 'LÃ¦seLotte',
  src: 'http://p-hold.com/200/200',
  image: 'http://p-hold.com/200/200'
};

describe('BookcaseTemplate', () => {
  test('circle of list items is shown', () => {
    const list = {
      _id: 'current-list-id',
      title: 'some title',
      description: 'some description',
      public: true,
      list: [
        createTestElement(1),
        createTestElement(2),
        createTestElement(3),
        createTestElement(4),
        createTestElement(5),
        createTestElement(6)
      ]
    };

    const tree = renderer
      .create(<BookcaseTemplate list={list} profile={profile} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
