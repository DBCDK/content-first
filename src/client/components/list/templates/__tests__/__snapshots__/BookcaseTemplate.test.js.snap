// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookcaseTemplate circle of list items is shown 1`] = `
<div
  className="bookcase-template"
>
  <BookcaseItem
    list={
      Object {
        "description": "some description",
        "descriptionImage": true,
        "id": "current-list-id",
        "list": Array [
          Object {
            "book": Object {
              "creator": "some creator1",
              "pid": "pid1",
              "title": "some title1",
            },
            "description": "some description1",
            "links": Object {
              "cover": "/cover1",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator2",
              "pid": "pid2",
              "title": "some title2",
            },
            "description": "some description2",
            "links": Object {
              "cover": "/cover2",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator3",
              "pid": "pid3",
              "title": "some title3",
            },
            "description": "some description3",
            "links": Object {
              "cover": "/cover3",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator4",
              "pid": "pid4",
              "title": "some title4",
            },
            "description": "some description4",
            "links": Object {
              "cover": "/cover4",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator5",
              "pid": "pid5",
              "title": "some title5",
            },
            "description": "some description5",
            "links": Object {
              "cover": "/cover5",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator6",
              "pid": "pid6",
              "title": "some title6",
            },
            "description": "some description6",
            "links": Object {
              "cover": "/cover6",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
        ],
        "public": true,
        "title": "some title",
      }
    }
    profile={
      Object {
        "image": "http://p-hold.com/200/200",
        "name": "LæseLotte",
        "src": "http://p-hold.com/200/200",
      }
    }
  />
  <SimpleList
    editButton={undefined}
    list={
      Object {
        "description": "some description",
        "descriptionImage": true,
        "id": "current-list-id",
        "list": Array [
          Object {
            "book": Object {
              "creator": "some creator1",
              "pid": "pid1",
              "title": "some title1",
            },
            "description": "some description1",
            "links": Object {
              "cover": "/cover1",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator2",
              "pid": "pid2",
              "title": "some title2",
            },
            "description": "some description2",
            "links": Object {
              "cover": "/cover2",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator3",
              "pid": "pid3",
              "title": "some title3",
            },
            "description": "some description3",
            "links": Object {
              "cover": "/cover3",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator4",
              "pid": "pid4",
              "title": "some title4",
            },
            "description": "some description4",
            "links": Object {
              "cover": "/cover4",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator5",
              "pid": "pid5",
              "title": "some title5",
            },
            "description": "some description5",
            "links": Object {
              "cover": "/cover5",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          Object {
            "book": Object {
              "creator": "some creator6",
              "pid": "pid6",
              "title": "some title6",
            },
            "description": "some description6",
            "links": Object {
              "cover": "/cover6",
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
        ],
        "public": true,
        "title": "some title",
      }
    }
    profile={
      Object {
        "image": "http://p-hold.com/200/200",
        "name": "LæseLotte",
        "src": "http://p-hold.com/200/200",
      }
    }
    profiles={undefined}
  />
</div>
`;
